src/System.Reflection.Extensions/src/System/Reflection/CustomAttributeExtensions.cs

```diff
@@ -0,0 +1,343 @@
+        }
+
+        public static IEnumerable<T> GetCustomAttributes<T>(this ParameterInfo element, bool inherit) where T : Attribute
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(typeof(T), inherit, skipTypeValidation: true);
+            return matches.Select(m => (T)(m.Instantiate()));
+        }
+
+        //==============================================================================================================
+        // This group determines whether the element has an associated custom attribute whose type matches or is a subclass
+        // of "attributeType".
+        //==============================================================================================================
+        public static bool IsDefined(this Assembly element, Type attributeType)
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(attributeType);
+            return matches.Any();
+        }
+
+        public static bool IsDefined(this Module element, Type attributeType)
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(attributeType);
+            return matches.Any();
+        }
+
+        public static bool IsDefined(this MemberInfo element, Type attributeType)
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(attributeType, inherit: true);
+            return matches.Any();
+        }
+
+        public static bool IsDefined(this MemberInfo element, Type attributeType, bool inherit)
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(attributeType, inherit);
+            return matches.Any();
+        }
+
+        public static bool IsDefined(this ParameterInfo element, Type attributeType)
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(attributeType, inherit: true);
+            return matches.Any();
+        }
+
+        public static bool IsDefined(this ParameterInfo element, Type attributeType, bool inherit)
+        {
+            IEnumerable<CustomAttributeData> matches = element.GetMatchingCustomAttributes(attributeType, inherit);
+            return matches.Any();
+        }
+
+
+        //==============================================================================================================================
+        // Helper for the GetCustomAttribute() family.
+        //==============================================================================================================================
+        private static T OneOrNull<T>(this IEnumerable<CustomAttributeData> results) where T : Attribute
+        {
+            IEnumerator<CustomAttributeData> enumerator = results.GetEnumerator();
+            if (!enumerator.MoveNext())
+                return null;
```

Comment:
The enumerator should be disposed of in a finally block before exiting the function.